### PR TITLE
API 8653 better abstractions

### PR DESCRIPTION
**Pull request by bschofield-va**
FilemanEntry now has convience methods to better support accessing internal/external field values.

```
Values v = entry.field(".01") // may return null
Optional<String> v = entry.internal("0.1") // .external(...)
Optional<Foo> f = entry.internal("0.1",Foo:fromString) // .external(...)
```

### CODE REVIEW

File: charon-models/src/main/java/gov/va/api/lighthouse/charon/models/lhslighthouserpcgateway/LhsLighthouseRpcGatewayResponse.java [hunk 1]
```diff
@@ -35,18 +40,83 @@
...
+      return fields().get(fieldNumber);
+    }
 
     /** Lazy Initialization. */
-    public List<FilemanEntry> results() {
-      if (results == null) {
-        return List.of();
+    public Map<String, Values> fields() {
+      if (fields == null) {
+        fields = new HashMap<>();
```
Comment (jhulbert-va) [inline charon-models/src/main/java/gov/va/api/lighthouse/charon/models/lhslighthouserpcgateway/LhsLighthouseRpcGatewayResponse.java:97]: `Map.of()`. I think this should be immutable.

File: charon-models/src/main/java/gov/va/api/lighthouse/charon/models/lhslighthouserpcgateway/PatientType.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package gov.va.api.lighthouse.charon.models.lhslighthouserpcgateway;
+
+public class PatientType {
+  public static final String FILE_NUMBER = "2";
+  public static final String INSURANCE_TYPE_SUBFILE = "#.3121";
```
Comment (jhulbert-va) [inline charon-models/src/main/java/gov/va/api/lighthouse/charon/models/lhslighthouserpcgateway/PatientType.java:5]: The field name should just be `INSURANCE_TYPE` based on the vista docs [here](https://vivian.worldvista.org/dox/Global_XkRQVA==.html)

File: charon-models/src/test/java/gov/va/api/lighthouse/charon/models/lhslighthouserpcgateway/LhsLighthouseRpcGatewayGetsManifestTest.java [hunk 3]
```diff
@@ -120,7 +120,7 @@ void requestAsDetails() {
         LhsLighthouseRpcGatewayGetsManifest.Request.builder()
             .file("2")
             .iens("1")
-            .fields(List.of(".01", ".3121*"))
+            .fields(List.of("#.01", ".3121*"))
```
Comment (jhulbert-va) [inline charon-models/src/test/java/gov/va/api/lighthouse/charon/models/lhslighthouserpcgateway/LhsLighthouseRpcGatewayGetsManifestTest.java:123]: Shouldn't `.3121` also have a `#`?

File: charon-models/src/test/java/gov/va/api/lighthouse/charon/models/lhslighthouserpcgateway/LhsLighthouseRpcGatewayListManifestTest.java [hunk 4]
```diff
@@ -123,7 +123,7 @@ void requestAsDetails() {
         LhsLighthouseRpcGatewayListManifest.Request.builder()
             .file("2")
             .iens(Optional.of("1"))
-            .fields(List.of(".01", ".3121*"))
+            .fields(List.of("#.01", ".3121*"))
```
Comment (jhulbert-va) [inline charon-models/src/test/java/gov/va/api/lighthouse/charon/models/lhslighthouserpcgateway/LhsLighthouseRpcGatewayListManifestTest.java:126]: Same.

File: charon-models/src/test/java/gov/va/api/lighthouse/charon/models/lhslighthouserpcgateway/LhsLighthouseRpcGatewayGetsManifestTest.java (hunk 3 above)
Comment (bschofield-va) [reply charon-models/src/test/java/gov/va/api/lighthouse/charon/models/lhslighthouserpcgateway/LhsLighthouseRpcGatewayGetsManifestTest.java:123]: No, it's showing that the `#` is optional.

File: charon-models/src/test/java/gov/va/api/lighthouse/charon/models/lhslighthouserpcgateway/LhsLighthouseRpcGatewayListManifestTest.java (hunk 4 above)
Comment (bschofield-va) [reply charon-models/src/test/java/gov/va/api/lighthouse/charon/models/lhslighthouserpcgateway/LhsLighthouseRpcGatewayListManifestTest.java:126]: same

File: charon-models/src/main/java/gov/va/api/lighthouse/charon/models/lhslighthouserpcgateway/LhsLighthouseRpcGatewayResponse.java (hunk 1 above)
Comment (bschofield-va) [reply charon-models/src/main/java/gov/va/api/lighthouse/charon/models/lhslighthouserpcgateway/LhsLighthouseRpcGatewayResponse.java:97]: `FilemanEntry` is `@Data`, therefore mutable. See `file` and `ien` fields.

**Review by eclendenning-va — APPROVED** [review]

File: charon-models/src/main/java/gov/va/api/lighthouse/charon/models/lhslighthouserpcgateway/LhsLighthouseRpcGatewayResponse.java (hunk 1 above)
Comment (jhulbert-va) [reply charon-models/src/main/java/gov/va/api/lighthouse/charon/models/lhslighthouserpcgateway/LhsLighthouseRpcGatewayResponse.java:97]: Alright, should this class use the `Value` annotation then? The values here shouldn't be muteable right? They're coming from Vista.

Comment (bschofield-va) [reply charon-models/src/main/java/gov/va/api/lighthouse/charon/models/lhslighthouserpcgateway/LhsLighthouseRpcGatewayResponse.java:97]: I don't think it really matters

**Pull request merged**